descriptions.md

```diff
@@ -1133,3 +1098,86 @@ Contains fields used when referencing a previous message.
 |---|---|---|---|---|
 | timeserial: String || | REX2a | A unique identifier indicating the message being interacted with. |
 | type: String || | REX2b | The reason for interacting with the previous message, for example, `com.ably.reaction` to indicate a reaction. |
+
+## class `BatchResult<T>`
+
+Contains information about the results of a batch operation.
+
+| Method / Property | Parameter | Returns | Spec | Description |
+|---|---|---|---|---|
+| successCount: Int ||| BAR2a | The number of successful operations in the request. |
+| failureCount: Int ||| BAR2b | The number of unsuccessful operations in the request. |
+| results: [T] ||| BAR2c | An array of results for the batch operation. |
+
+## class `BatchPublishSpec`
+
+Describes the messages that should be published by a batch publish operation, and the channels to which they should be published.
+
+| Method / Property | Parameter | Returns | Spec | Description |
+|---|---|---|---|---|
+| channels: [String] ||| BSP2a | The names of the channels to publish the `messages` to. |
+| messages: [Message] ||| BSP2b | An array of [`Message`]{@link Message} objects. |
+
+## class `BatchPublishSuccessResult`
+
+Contains information about the result of successful publishes to a channel requested by a single [`BatchPublishSpec`]{@link BatchPublishSpec}.
+
+| Method / Property | Parameter | Returns | Spec | Description |
+|---|---|---|---|---|
+| channel: String ||| BPR2a | The name of the channel the message(s) was published to. |
+| messageId: String ||| BPR2b | A unique ID prefixed to each published message. |
```

Comment:
I think we need to say something about how to interpret this property, based on https://github.com/ably/specification/pull/139#discussion_r1184224904. Here's my badly-worded suggestion, perhaps @m-hulbert can help with the wording?

```suggestion
| messageId: String ||| BPR2b | An identifier shared by all of the messages published as a result of this `BatchPublishSpec` request. Clients receiving the message at index `i` of the `BatchPublishSpec.messages` array will see a `Message.id` property of `${messageId}.${i}`. |
```

Reply:
I've updated it to "A unique ID prefixed to the [`id`]{@link Message#id} of each published message."